### ISSUE TITLE
Gentoo package

### ISSUE THREAD

**Issue by xgqt** [justinethier/cyclone#457]
Hello! 

I thought I'd let you know that I packaged Cyclone for the Gentoo GURU repository

https://gitweb.gentoo.org/repo/proj/guru.git/tree/dev-scheme/cyclone/cyclone-9999.ebuild?h=dev&id=e19e9244b23a20ecc0b857d055ca53f3184408d4

Do you have any suggestions or concerns about the ebuild script?

**Comment by xgqt** [comment]
Alternative link: https://github.com/gentoo/guru/blob/dev/dev-scheme/cyclone/cyclone-9999.ebuild

**Comment by justinethier** [comment]
Hey, this is great! Thank you for making this happen :)

Without having a Gentoo system handy here are my initial comments:

- The `-rdynamic` option is only used in our build scripts for building with msys2 on Windows. I suspect this option is not required.
- Alternatively, the `-O2` option is very useful to increase performance in the compiled code. Maybe this is already incorporated into the build, though, just not explicitly in this file. I do suggest making sure the build is setup to pass this option to the C compiler.

**Comment by xgqt** [comment]
>     * The `-rdynamic` option is only used in our build scripts for building with msys2 on Windows. I suspect this option is not required.

I see


>     * Alternatively, the `-O2` option is very useful to increase performance in the compiled code. Maybe this is already incorporated into the build, though, just not explicitly in this file. I do suggest making sure the build is setup to pass this option to the C compiler.

-march=..., -O... and -f... would be already set by user in CFLAGS

btw, assuming user uses CFLAGS="-march=native -O3", in the src_compile the variables will equal to:
CYC_GCC_OPT_FLAGS="-march=native -O3"
CFLAGS="-march=native -O3 -fPIC -Iinclude"

**Comment by xgqt** [comment]
> append-ldflags -L. -Wl,--export-dynamic

I think -Wl,--export-dynamic should already be set in user's LDFLAGS but force-append it anyway

**Comment by xgqt** [comment]
I think -Wl,--export-dynamic should be already set in user's LDFLAGS. Do you think I should remove it?
![image](https://user-images.githubusercontent.com/47485207/118450304-92b37d00-b6e3-11eb-82c3-58f000cdbebe.png)
^ duplicated -Wl,--export-dynamic

**Comment by xgqt** [comment]
On Gentoo tommath can be installed as system lib - `dev-libs/libtommath`. Maybe you could add option to build with sysem's tommath.

**Comment by justinethier** [comment]
Yes, `export-dynamic` is already set, no need to set again in the ebuild file.

It is convenient to have tommath bundled so that we remove an installation dependency. This also avoids any potential issue with a version of tommath (EG: a new release) on the system not working with the version of Cyclone. That said, if you think there is value to using the system version I can put in a ticket to allow the option for tommath.

**Comment by xgqt** [comment]
> This also avoids any potential issue with a version of tommath (EG: a new release) on the system not working with the version of Cyclone.

On Gentoo we can avoid this in 2 ways:
1) pin the dependency to a specific version `?dev-libs/libtommath-1.2.0`, where ? will be one of `>`, `=`, `<` which can describe if we need a dependency higher, equal or lower than specified package version
2) USE flag `system-tommath` can be added to give user control whether they want to use the bundled one or a system one


> if you think there is value to using the system version I can put in a ticket to allow the option for tommath.

Oh yes, there is definitely a benefit to system versions

**Comment by justinethier** [comment]
Good deal, I've added https://github.com/justinethier/cyclone/issues/458 to address this.

Do you see issues with packaging 0.28.0 in the meantime?

**Comment by xgqt** [comment]
> Do you see issues with packaging 0.28.0 in the meantime?

No. I played with it a bit and noticed `PREFIX` also has to be exported during build, not only install.
Fixed it with this commit: https://gitweb.gentoo.org/repo/proj/guru.git/commit/?h=dev&id=f63e7f8a8f5af1eae35343fc4da3486213175a81

**Comment by xgqt** [comment]
> > Do you see issues with packaging 0.28.0 in the meantime?
> 
> No, not at the moment. I played with it a bit and noticed `PREFIX` also has to be exported during build, not only install.
> Fixed it with this commit: https://gitweb.gentoo.org/repo/proj/guru.git/commit/?h=dev&id=f63e7f8a8f5af1eae35343fc4da3486213175a81

forgot to say I noticed this because `import` was looking for libs in /usr/lib instead of /usr

**Comment by xgqt** [comment]
I'd like to announce that cyclone is officially in the Gentoo main repository :D https://packages.gentoo.org/packages/dev-scheme/cyclone

**Comment by justinethier** [comment]
Great news! Thanks for the update and for your efforts here @xgqt 😄
